app/controllers/subscription/SubscriptionFailureController.scala

```diff
@@ -18,23 +18,38 @@ package controllers.subscription
 
 import config.FrontendAppConfig
 import controllers.actions.IdentifierAction
+import models._
+import pages.PlrReferencePage
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.EmptyStateHomepageView
 import views.html.errors.SubscriptionFailureView
 
 import javax.inject.Inject
+import scala.concurrent.ExecutionContext
 
 class SubscriptionFailureController @Inject() (
   identify:                 IdentifierAction,
   val controllerComponents: MessagesControllerComponents,
-  view:                     SubscriptionFailureView
-)(implicit appConfig:       FrontendAppConfig)
+  view:                     SubscriptionFailureView,
```

Comment:
Can you use `subscriptionFailureView` instead of just `view`? Makes the code more easy to read and understand which view we are using.

```suggestion
  subscriptionFailureView:                     SubscriptionFailureView,
```